packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -177,11 +193,16 @@ function NavigationMenuSelector( {
 					{ canUserCreateNavigationMenu && (
 						<MenuGroup label={ __( 'Tools' ) }>
 							<MenuItem
-								disabled={ isCreatingMenu }
+								disabled={
+									isUpdatingMenuRef ||
+									! hasResolvedNavigationMenus
+								}
 								onClick={ () => {
-									onClose();
-									onCreateNew();
-									setIsCreatingMenu( true );
+									setIsUpdatingMenuRef( true );
+									onCreateNew().then( () => {
+										setIsUpdatingMenuRef( false );
```

Comment:
Removing this line allows the 'should handle disabled state of the create menu button during the creation process' test to pass. I think it's because the promise onCreateNew is resolving to `false` before the unit test can check for the disabled state.

I believe [this is the line in the unit test](https://github.com/WordPress/gutenberg/pull/59630/files?diff=split&w=1#diff-5acf9be3bfd5e32ba550339fe797b11880fdb50271e88810c2e879ded5f9cddbR278-R282) that both sets the button to be disabled and then resolves to no longer be disabled, so the unit test doesn't even "see" the state of it being disabled.